packages/redux/src/users/returns/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,178 @@
+import * as actionTypes from '../actionTypes';
+import { BlackoutError, toBlackoutError } from '@farfetch/blackout-client';
+import { mockUserReturnsResponse } from 'tests/__fixtures__/users';
+import reducer, * as fromReducer from '../reducer';
+import type { UserReturnsState } from '../types';
+
+let initialState: UserReturnsState;
+
+describe('returns reducers', () => {
+  beforeEach(() => {
+    initialState = fromReducer.INITIAL_STATE;
+  });
+
+  describe('result() reducer', () => {
+    it('should return the initial state', () => {
+      const newState = reducer(undefined, { type: 'INIT' }).result;
+
+      expect(newState).toBe(initialState.result);
+      expect(newState).toBeNull();
+    });
+
+    it('should handle FETCH_USER_RETURNS_SUCCESS action type', () => {
+      const newResult = {
+        result: {
+          number: 1,
+          totalPages: 1,
+          totalItems: 1,
+          entries: [{ id: '111' }, { id: '222' }],
+        },
+      };
+      const previousState = {
+        error: null,
+        isLoading: false,
+        result: {
+          number: 1,
+          totalPages: 1,
+          totalItems: 1,
+          entries: [],
+        },
+      };
+
+      const newState = reducer(previousState, {
+        payload: newResult,
+        type: actionTypes.FETCH_USER_RETURNS_SUCCESS,
+      });
+
+      expect(newState.result).toEqual(newResult);
+    });
+
+    it('should handle other actions by returning the previous state', () => {
+      const state = {
+        result: mockUserReturnsResponse,
+        error: null,
+        isLoading: false,
+      };
+
+      const randomActionWithResult = {
+        type: 'random',
+        result: [{ id: '222' }],
+      };
+
+      expect(reducer(state, randomActionWithResult).result).toBe(state.result);
+    });
+  });
+
+  describe('error() reducer', () => {
+    it('should return the initial state', () => {
+      const state = reducer(undefined, { type: 'INIT' }).error;
+
+      expect(state).toBe(initialState.error);
+      expect(state).toBeNull();
+    });
+
+    // Error value on FAILURE
+    it.each([actionTypes.FETCH_USER_RETURNS_FAILURE])(
+      'should handle %s action type',
+      actionType => {
+        const error = 'foo';
+        const reducerResult = reducer(undefined, {
```

Comment:
You should test with a state that is more representative of when this action is dispatched. 
**For example:** 

`const mockState = { ...initialState, error: null };`

Pls, review the following tests, because with undefined we don't know the state before it was dispatched

Reply:
done 👍🏻

Reply:
Did you fix the following tests as well?

Reply:
yep, all cases where reviewed.